src/ITensorNetworks/models/hamiltonians.jl

```diff
@@ -0,0 +1,75 @@
+using ITensors
+
+struct LocalMPO
+  mpo::MPO
+  coord1::Pair{<:Integer,<:Integer}
+  coord2::Pair{<:Integer,<:Integer}
+end
+
+# Transverse field
+# The critical point is h = 1.0
+# This is the most challenging part of the model for DMRG
+function mpo(::Model"tfim", sites::Matrix{<:Index}; h::Float64)
+  Ny, Nx = size(sites)
+  sites_vec = reshape(sites, Nx * Ny)
+  lattice = square_lattice(Nx, Ny; yperiodic=false)
+
+  opsum = OpSum()
+  for b in lattice
+    opsum += -1, "X", b.s1, "X", b.s2
+  end
+  for i in 1:(Nx * Ny)
+    opsum += h, "Z", i
+  end
+  return MPO(opsum, sites_vec)
+end
+
+# Get the local hamiltonian term of a 2D grid
+function localham_term(::Model"tfim", sites::Matrix{<:Index}, bond; h::Float64)
+  Ny, Nx = size(sites)
+  sites_vec = reshape(sites, Nx * Ny)
+  n1, n2 = bond.s1, bond.s2
+  opsum = OpSum()
+  opsum += -1, "X", 1, "X", 2
+  if n2 == n1 + 1
+    opsum += h, "Z", 1
+  end
+  if n2 == n1 + 1 && n2 % Ny == 0
+    opsum += h, "Z", 2
+  end
+  mpo = MPO(opsum, sites_vec[[n1, n2]])
+  coord1 = ((n1 - 1) % Ny + 1) => trunc(Int, (n1 - 1) / Ny) + 1
+  coord2 = ((n2 - 1) % Ny + 1) => trunc(Int, (n2 - 1) / Ny) + 1
+  return LocalMPO(mpo, coord1, coord2)
+end
+
+# Return a list of LocalMPO
+function localham(m::Model, sites; kwargs...)
+  Ny, Nx = size(sites)
+  lattice = square_lattice(Nx, Ny; yperiodic=false)
```

Comment:
For this, it would be nice to make use of the `Square` lattice that I defined: https://github.com/ITensor/ITensorNetworkAD.jl/blob/main/src/ITensorNetworks/tensor_networks.jl#L33-L40.

The `square_lattice` in ITensors.jl is more geared towards a 1-dimensional "snake" pattern through a 2-dimensional lattice. The `Square` lattice type I made is more directly suited for 2 dimensions (for example, there are iterators over bonds/edges which are of the form `((2, 2), (2, 3))`, so actual 2-dimensional coordinates for the sites). Let me know if you have questions about how to use it or if there are issues.

The `localham_term` function would need to be adapted to those new types of bonds.

Reply:
Again, this is not the most important thing right now as long as the current functionality works, but making you aware of it and we can leave it for a future issue.

Reply:
For this I will open an issue and fix this later